go/src/meguca/templates/article.go

```diff
@@ -115,14 +122,88 @@ func extractBacklinks(cap int, threads ...common.Thread) backlinks {
 	return bls
 }
 
-// Returns the stringified n + the plural or singular word from the language
-// by index word
+// Returns 3 mod-log related messages by post ID
+func parseModLog(p common.Post) (any bool, attrs string, msgs string) {
+	ln := lang.Get().Common.Posts
+	log := GetPostModLog(p.ID)
+
+	for i, val := range [3]bool{p.Banned, p.Deleted, p.MeidoVision} {
+		if val {
+			switch i {
+			case 0:
+				any = true
+				attrs = " banned"
+				msgs = ln["banned"]
+
+				if log[i].ID != 0 {
+					msgs += ` BY "` + log[i].By + `" FOR ` +
+						strings.ToUpper(secondsToTime(float64(log[i].Length))) +
+						": " + log[i].Reason
+				}
+			case 1:
+				if log[i].ID != 0 {
+					if (any) {
+						msgs += "<br>"
+					}
+		
+					any = true
+					attrs += " deleted"
+					msgs += ln["deleted"] + ` BY "` + log[i].By + `"`
+				}
+			case 2:
+				if (any) {
+					msgs += "<br>"
+				}
+
+				any = true
+				attrs += " meido-vision"
+				msgs += ln["meidoVision"]
+				
+				if log[i].ID != 0 {
+					msgs += ` BY "` + log[i].By + `"`
+				}
+			}
+		}
+	}
+
+	return
+}
+
+// Returns human readable time
+func secondsToTime(s float64) string {
+	time := math.Floor(s) / 60
+	divide := [4]float64{60, 24, 30, 12}
```

Comment:
Note: you can use `[...]float64` for Go to infer the length of the array automatically.

Reply:
I'd rather stick to a static declaration, unless you really really want to use `[...]` for some reason.